### PR TITLE
Clean up CustomRequestLog and fix the handling of the %u code.

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Replacing PR #4364.

The logic for the parameter of the `%u` code was wrong, and has been fixed so that:
```
   %u -> logRequestAuthentication
%{d}u -> logRequestAuthenticationWithDeferred
```

I have also explained in the javadoc that `%{d}u` is equivalent to is equivalent to `HttpServletRequest.getRemoteUser()`.

Also doing a few minor cleanups to clear warnings from intellij.

### CODE REVIEW

**Comment by sbordet** [comment]
@lachlan-roberts where is the test case?

**Comment by joakime** [comment]
The 2 existing `CustomRequestLogTest` classes do not appear to have tests for this `%u` format parameter.

File: tests/test-integration/src/test/java/org/eclipse/jetty/test/CustomRequestLogTest.java [hunk 1]
```diff
@@ -88,24 +102,70 @@ void testHandlerServerStart(String formatString) throws Exception
...
+        String authHeader = HttpHeader.AUTHORIZATION + ": Basic " + Base64.getEncoder().encodeToString("username:password".getBytes());
+        testHandlerServerStart("%u %{d}u");
+
+        _connector.getResponse("GET / HTTP/1.0\n\n\n");
+        String log = _entries.poll(5, TimeUnit.SECONDS);
+        assertThat(log, is("- -"));
+
+        _connector.getResponse("GET / HTTP/1.0\n" + authHeader + "\n\n\n");
+        log = _entries.poll(5, TimeUnit.SECONDS);
+        assertThat(log, is("- ?username"));
```
Comment (lachlan-roberts) [inline tests/test-integration/src/test/java/org/eclipse/jetty/test/CustomRequestLogTest.java]: Do you think we should be appending this `?` before the remote user if it was deferred? Seems like a strange thing to do.

**Comment by lachlan-roberts** [comment]
Failures are all unrelated HTTP/2 Client tests.

File: tests/test-integration/src/test/java/org/eclipse/jetty/test/CustomRequestLogTest.java (hunk 1 above)
Comment (gregw) [reply tests/test-integration/src/test/java/org/eclipse/jetty/test/CustomRequestLogTest.java]: Probably not.  Why not have another format %u{D} that either does have the '?'

**Review by gregw — APPROVED** [review]
LGTM

**Review by sbordet — APPROVED** [review]

**Pull request merged**